### ISSUE TITLE
7$: Warn when deleting variable

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#251]
In the following code, `delete x;` seems to have no effect:
```js
let x = 3;
delete x;
console.log(x);
```
Have quick-lint-js report a warning saying that `delete` has no effect.

**Comment by singalhimanshu** [comment]
I claim this for-hire task. I expect payment after I complete this task. I will email the quick-lint-js team if I am assigned this task.

**Comment by singalhimanshu** [comment]
What files should I look in for this issue?

**Comment by strager** [comment]
> What files should I look in for this issue?

parse.cpp, maybe parse.h.
